subsys/bluetooth/services/ancs_attr_parser.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2016 - 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/* Disclaimer: This client implementation of the Apple Notification Center Service can and will be changed at any time by Nordic Semiconductor ASA.
+ * Server implementations such as the ones found in iOS can be changed at any time by Apple and may cause this client implementation to stop working.
+ */
+
+#include <sys/byteorder.h>
+#include <bluetooth/services/ancs_c.h>
+#include <bluetooth/services/ancs_attr_parser.h>
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(ancs_c, CONFIG_BT_GATT_ANCS_C_LOG_LEVEL);
+
+static bool all_req_attrs_parsed(struct bt_gatt_ancs_c *ancs_c)
+{
+	if (ancs_c->parse_info.expected_number_of_attrs == 0) {
+		return true;
+	}
+	return false;
+}
+
+static bool attr_is_requested(struct bt_gatt_ancs_c *ancs_c,
+			      bt_gatt_ancs_c_attr_t attr)
+{
+	if (ancs_c->parse_info.p_attr_list[attr.attr_id].get == true) {
+		return true;
+	}
+	return false;
+}
+
+/**@brief Function for parsing command id and notification id.
+ *        Used in the @ref parse_get_notif_attrs_response state machine.
+ *
+ * @details UID and command ID will be received only once at the beginning of the first
+ *          GATTC notification of a new attribute request for a given iOS notification.
+ *
+ * @param[in] ancs_c     Pointer to an ANCS instance to which the event belongs.
+ * @param[in] p_data_src Pointer to data that was received from the Notification Provider.
+ * @param[in] index      Pointer to an index that helps us keep track of the current data to be parsed.
+ *
+ * @return The next parse state.
+ */
+static bt_gatt_ancs_c_parse_state_t
```

Comment:
please use line-break in argument list instead.

Reply:
Done